backend/tests/test_transactions.py

```diff
@@ -0,0 +1,415 @@
+            "amount": amount,
+            "category": category,
+            "notes": notes,
+            "account_id": account_id,
+            "username": username,
+        },
+    )
+    new_date = "0001-01-01"
+    res = client.post(
+        "/transaction",
+        json={
+            "transaction_id": init_res.json["transaction"]["transaction_id"],
+            "date": new_date,
+            "amount": amount + 5000,
+            "category": category + 2,
+            "notes": notes + " and some extra notes",
+            "account_id": post_res.json["account"]["account_id"],
+            "username": username,
+        },
+    )
+    assert res.status_code == 200
+    assert len(res.json["transaction"]["transaction_id"]) == 36
+    assert res.json["transaction"]["date"] == new_date
+    assert res.json["transaction"]["amount"] == amount + 5000
+    assert res.json["transaction"]["category"] == category + 2
+    assert res.json["transaction"]["notes"] == notes + " and some extra notes"
+    assert (
+        res.json["transaction"]["account_id"] == post_res.json["account"]["account_id"]
+    )
+    assert res.json["transaction"]["username"] == username
+
+
+@pytest.mark.parametrize(
+    "transaction_id, date, amount, category, notes, account_id, username",
+    TEST_TRANSACTION_DATA,
+)
+def test_get_transaction_success(
+    client, transaction_id, date, amount, category, notes, account_id, username
+):
+    client.post("/user", json={"username": username, "first_name": "", "last_name": ""})
+    post_res = client.post(
+        "/transaction",
+        json={
+            "transaction_id": transaction_id,
+            "date": date,
+            "amount": amount,
+            "category": category,
+            "notes": notes,
+            "account_id": account_id,
+            "username": username,
+        },
+    )
+    trans_id = post_res.json["transaction"]["transaction_id"]
+
+    res = client.get(f"/transaction?transaction_id={trans_id}&username={username}")
+
+    assert res.status_code == 200
+    assert len(res.json["transaction"]["transaction_id"]) == 36
+    assert res.json["transaction"]["transaction_id"] == trans_id
+    assert res.json["transaction"]["date"] == date
+    assert res.json["transaction"]["amount"] == amount
+    assert res.json["transaction"]["category"] == category
+    assert res.json["transaction"]["notes"] == notes
+    assert res.json["transaction"]["account_id"] == account_id
+    assert res.json["transaction"]["username"] == username
+
+
+@pytest.mark.parametrize(
+    "transaction_id, date, amount, category, notes, account_id, username",
+    TEST_EXISTING_TRANSACTION_DATA,
+)
+def test_get_transaction_failure(
+    client, transaction_id, date, amount, category, notes, account_id, username
+):
+    client.post("/user", json={"username": username, "first_name": "", "last_name": ""})
+    res = client.get(
+        f"/transaction?transaction_id={transaction_id}&username={username}"
```

Comment:
Was thinking about moving it into another endpoint, but it's already set up to return all transactions (same with the account endpoint) so I'll just add it. Might be better elsewhere to separate them for clarity but since I'm the only one using it it's probably fine.